packages/portfolio-deploy/src/vstorage-prune.core.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file core eval to prune (presumably outdated) vstorage nodes.
+ *
+ * Use `vstorage-prune.build.js` to configure which nodes to prune.
+ */
+
+/**
+ * @import {ChainStoragePresent} from './chain-info.core'
+ */
+
+import { makeTracer } from '@agoric/internal';
+import { E } from '@endo/eventual-send';
+
+const trace = makeTracer('VSP');
+
+/**
+ * @typedef {{
+ *   toPrune: Record<string, string[]>
+ * }} PruneOpts
+ */
+
+/**
+ * @param {BootstrapPowers & ChainStoragePresent} permitted
+ * @param {{options: PruneOpts}} config
+ */
+export const pruneVStorage = async (permitted, { options }) => {
+  const { toPrune } = options;
+  trace('toPrune', toPrune);
+  const { chainStorage } = permitted.consume;
+
+  /** @param {string[]} path */
+  const makePathNode = path => {
+    let node = chainStorage;
+    for (const segment of path) {
+      node = E(node).makeChildNode(segment);
+    }
+    return node;
+  };
+
+  await null;
+  for (const [parent, children] of Object.entries(toPrune)) {
+    const segments = parent.replace(/^published\./, '').split('.');
+    const parentNode = makePathNode(segments);
+    trace('pruning', parent, children);
+    await Promise.all(
+      children.map(k =>
+        E(E(parentNode).makeChildNode(k, { sequence: false })).setValue(''),
+      ),
+    );
```

Comment:
I'm a little bit hesitant to see a `Promise.all(arr.map(...))` without more explicit propagation of the context causing a rejection.  And, maybe it's worth doing a best-effort deletion (i.e. log but don't fail if the deletion fails).

No need to do this, though.  If you have to update the PR before merging, consider adding some comments where we could be more lenient (like considering `Promise.allSettled` instead of `Promise.all`, and logging which kid throws when `setValue`d).

Reply:
I went with `allSettled`, but in retrospect I suppose it would be simpler to just catch inside the `Promise.all`:

```js
    await Promise.all(
      children.map(async child => {
        await null;
        // on failure, trace rather than aborting the whole job
        try {
          const nodeP = E(parentNode).makeChildNode(child, { sequence: false });
          await E(nodeP).setValue('');
        } catch (err) {
          trace('rejected:', parent, child, err);
        }
      }),
    );
```

Reply:
OTOH, I'm not aware of any failure modes of `makeChildNode` nor `setValue`, except perhaps vat death.